x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -27,26 +38,115 @@ export const calculateEndpointAuthz = (
-    ? fleetAuthz.packagePrivileges?.endpoint?.actions?.writeHostIsolation?.executePackageAction ||
-      false
-    : hasEndpointManagementAccess;
-  const canWriteProcessOperations = isEndpointRbacEnabled
-    ? fleetAuthz.packagePrivileges?.endpoint?.actions?.writeProcessOperations
-        ?.executePackageAction || false
-    : hasEndpointManagementAccess;
+  const canWritePolicyManagement = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writePolicyManagement'
+  );
+  const canReadPolicyManagement = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readPolicyManagement'
+  );
+  const canWriteActionsLogManagement = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeActionsLogManagement'
+  );
+  const canReadActionsLogManagement = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readActionsLogManagement'
+  );
+  const canIsolateHost = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeHostIsolation'
+  );
+  const canWriteProcessOperations = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeProcessOperations'
+  );
+  const canWriteTrustedApplications = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeTrustedApplications'
+  );
+  const canReadTrustedApplications = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readTrustedApplications'
+  );
+  const canWriteHostIsolationExceptions = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeHostIsolationExceptions'
+  );
+  const canReadHostIsolationExceptions = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readHostIsolationExceptions'
+  );
+  const canWriteBlocklist = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeBlocklist'
+  );
+  const canReadBlocklist = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readBlocklist'
+  );
+  const canWriteEventFilters = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'writeEventFilters'
+  );
+  const canReadEventFilters = hasPermission(
+    fleetAuthz,
+    isEndpointRbacEnabled,
+    hasEndpointManagementAccess,
+    'readEventFilters'
+  );
 
   return {
     canAccessFleet: fleetAuthz?.fleet.all ?? userRoles.includes('superuser'),
     canAccessEndpointManagement: hasEndpointManagementAccess,
     canCreateArtifactsByPolicy: hasEndpointManagementAccess && isPlatinumPlusLicense,
+    canWritePolicyManagement: canWritePolicyManagement,
```

Comment:
we don't have a consolidated place for all the licensing requirements so I didn't add them here. we should make sure to add this as we implement each one.